infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/ElementListing.kt

```diff
@@ -122,50 +130,62 @@ fun toElementListing(
 )
 
 fun elementListing(
-    geocodingContext: GeocodingContext?,
+    context: GeocodingContext?,
+    getTransformation: (srid: Srid) -> Transformation,
     planId: DomainId<GeometryPlan>,
     fileName: FileName,
     units: GeometryUnits,
     trackNumberId: IntId<TrackLayoutTrackNumber>?,
     trackNumberDescription: PlanElementName?,
     alignment: GeometryAlignment,
     element: GeometryElement,
-) = ElementListing(
-    planId = planId,
-    fileName = fileName,
-    coordinateSystemSrid = units.coordinateSystemSrid,
-    coordinateSystemName = units.coordinateSystemName,
-    trackNumberId = trackNumberId,
-    trackNumberDescription = trackNumberDescription,
-    alignmentId = alignment.id,
-    alignmentName = alignment.name,
-    elementId = element.id,
-    elementType = element.type,
-    lengthMeters = round(element.calculatedLength, LENGTH_DECIMALS),
-    start = getStartLocation(geocodingContext, alignment, element),
-    end = getEndLocation(geocodingContext, alignment, element),
-)
-
-fun getStartLocation(geocodingContext: GeocodingContext?, alignment: GeometryAlignment, element: GeometryElement) =
-    ElementLocation(
-        coordinate = element.start,
-        address = getAddress(geocodingContext, element.start),
-        directionGrads = getDirectionGrads(element.startDirectionRads),
-        radiusMeters = getStartRadius(element),
-        cant = getStartCant(alignment, element)
+) = units.coordinateSystemSrid?.let(getTransformation).let { transformation ->
```

Comment:
Jos Transformationia ei jostain syystä saada haettua, niin se johtaa `start`in ja `end`in `address`-arvojen nulliutumiseen. Onko se ok, vai pitäisikö tässä nostaa kädet pystyyn ja esim. heittää poikkeus?

Reply:
Mietin tuota itsekin mutta päädyin nyt tähän. Se on tietty aika omituista jos geometria on onnistuttu linkittämään ja kuitenkaan ei osata muuntaa sen koordinaatteja, joten poikkeuskin olisi tavallaan ok. Mutta ajattelin että se luo tarpeettoman kytkennän linkityksen toimintasekvenssin ja tämän välille. Jos koordinaatteja ei voi muuntaa niin osoitteita ei voi myöskään hakea -> osoite jää pois :shrug: